theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -1931,3 +1931,129 @@ def test_interface(self):
         fun = theano.function([x_sym, dfilter_sym, pfilter_sym], sep_op, mode='FAST_RUN')
         top = fun(x[:, :, :3, :3], depthwise_filter, pointwise_filter)
         utt.assert_allclose(top, precomp_output)
+
+
+class TestUnsharedConv(unittest.TestCase):
+    conv2d = theano.tensor.nnet.abstract_conv.AbstractConv2d
+    conv2d_gradw = theano.tensor.nnet.abstract_conv.AbstractConv2d_gradWeights
+    conv2d_gradi = theano.tensor.nnet.abstract_conv.AbstractConv2d_gradInputs
+    conv2d_op = theano.tensor.nnet.abstract_conv.AbstractConv2d
+    conv2d_gradw_op = theano.tensor.nnet.abstract_conv.AbstractConv2d_gradWeights
+    conv2d_gradi_op = theano.tensor.nnet.abstract_conv.AbstractConv2d_gradInputs
+
+    def setUp(self):
+        self.mode = theano.compile.mode.Mode(optimizer='None')
+
+        self.img_shape = [(2, 6, 5, 5), (1, 4, 7, 5), (1, 8, 5, 3), (1, 4, 7, 7)]
+        self.kern_shape = [(2, 3, 3, 6, 3, 3), (2, 6, 4, 4, 5, 3), (4, 2, 1, 2, 3, 3), (4, 4, 5, 2, 5, 3)]
+        self.topgrad_shape = [(2, 2, 3, 3), (1, 2, 6, 4), (1, 4, 2, 1), (1, 4, 4, 5)]
+        self.border_mode = ['valid', 'full', 'valid', 'full']
+        self.subsample = [(1, 1), (2, 2), (2, 1), (3, 2)]
+        self.filter_dilation = (1, 1)
+        self.num_groups = [1, 1, 4, 2]
+
+        self.ref_mode = 'FAST_RUN'
+        if theano.config.cxx == "":
+            raise SkipTest("CorrMM needs cxx")
+
+    def test_fwd(self):
+        tensor6 = theano.tensor.TensorType(theano.config.floatX, (False,) * 6)
+        img_sym = theano.tensor.tensor4('img')
+        kern_sym = tensor6('kern')
+        ref_kern_sym = theano.tensor.tensor4('ref_kern')
+
+        for imshp, kshp, mode, sub, groups in zip(self.img_shape, self.kern_shape, self.border_mode,
+                                                  self.subsample, self.num_groups):
+            img = np.random.random(imshp).astype(theano.config.floatX)
+            kern = np.random.random(kshp).astype(theano.config.floatX)
+
+            unshared_conv_op = self.conv2d(border_mode=mode, subsample=sub,
+                                           filter_dilation=self.filter_dilation,
+                                           num_groups=groups, unshared=True)
+            unshared_out_sym = unshared_conv_op(img_sym, kern_sym)
+            unshared_func = theano.function([img_sym, kern_sym], unshared_out_sym, mode=self.mode)
```

Comment:
Here you need to check that you're using the right op:
```python
            assert any([isinstance(node.op, self.conv2d_op)
                       for node in unshared_func.maker.fgraph.toposort()])
```
and something similar for `test_gradweight` and `test_gradinput`.